src/runtime/semaphore_wrapper.cc

```diff
@@ -50,6 +50,11 @@ void Semaphore::Wait() {
 }
 
 bool Semaphore::TimedWait(int timeout) {
+  // sem_timedwait does not exist in Mac OS.
+#ifdef __APPLE__
+#warning Timeout is not supported in semaphore's wait on Mac OS.
```

Comment:
Please add runtime warning instead of compilation warning. It won't be seen by users. Something like `DLOG(WARNING) << ...`